lab_1/main.py

```diff
@@ -64,31 +67,51 @@ def calculate_frequencies(tokens: list) -> dict:
 
 
 def get_top_n_words(freq_dict: dict, top_n: int) -> list:
-    k = 0
-    for i in range(len(string)):  # проверка строки на корректность
-        if 48 <= ord(' '.join(string)[i]) <= 57 or (128 <= ord(' '.join(string)[i]) <= 175 or 224 <= ord(' '.join(string)[i]) <= 243):
-            k += 1
-            break
-
-    if k != 0:  # если некорректные токены
+    """
+    Returns the most common words
+    :param freq_dict: a dictionary with frequencies
+    :param top_n: a number of the most common words to return
+    :return: a list of the most common words
+    e.g. tokens = ['weather', 'sunny', 'man', 'happy', 'and', 'dog', 'happy']
+    top_n = 1
+    --> ['happy']
+    """
+    if not isinstance(freq_dict, dict) or not isinstance(top_n, int):
         return []
     else:
         top_n = []
-        frequency = []
-        number = 0
-        for key, value in dictionary.items():
-            frequency.append([value, key])
-        frequency.sort(reverse=True)
-        if n <= len(frequency):
-            for element in frequency:
-                if number < n:
-                    top_n.append(element[1])
-                    number += 1
-            return top_n
+    frequency = []
+    number = 0
+    for key, value in dictionary.items():
+        frequency.append([value, key])
+    frequency.sort(reverse=True)
+    if n <= len(frequency):
+        for element in frequency:
+            if number < n:
+            top_n.append(element[1])
```

Comment:
tests fail here. Add \tab

if number < n:
____top_n...

___ – is space

Reply:
Thank you! But there are 4 tests crashing because of 21st line, it gives IndexError: string index out of range. How can I fix that?

Reply:
send me screenshots, please